samples/javascript_nodejs/11.qnamaker/index.js

```diff
@@ -48,20 +47,41 @@ const qnaEndpointSettings = {
     knowledgeBaseId: qnaConfig.kbId,
     endpointKey: qnaConfig.endpointKey,
     host: qnaConfig.hostname
-}
+};
 
 // Create adapter. See https://aka.ms/about-bot-adapter to learn more about adapters.
 const adapter = new BotFrameworkAdapter({
     appId: endpointConfig.appId || process.env.MicrosoftAppId,
     appPassword: endpointConfig.appPassword || process.env.MicrosoftAppPassword
 });
 
-// Create the bot that will handle incoming messages.
-const qnaMakerBot = new QnAMakerBot(qnaEndpointSettings, {});
+// Catch-all for errors.
+adapter.onTurnError = async (turnContext, error) => {
```

Comment:
onTurnError needs to use the load, clear, save pattern.  


    // Clear out state
    await conversationState.load(context);
    await conversationState.clear(context);
    // Save state changes.
    await conversationState.saveChanges(context);

Reply:
Sample does not use `ConversationState` which is why it's missing the state clearing mechanisms.